src/video/cocoa/cocoa_wnd.mm

```diff
@@ -1269,8 +1272,36 @@ - (void)windowDidEnterFullScreen:(NSNotification *)aNotification
 /** Screen the window is on changed. */
 - (void)windowDidChangeBackingProperties:(NSNotification *)notification
 {
+	ZoomLevel newSuggestedZoomLevel = driver->GetSuggestedUIZoom();
+	if (suggestedZoomLevel != newSuggestedZoomLevel) {
+		GfxClearSpriteCache();
+		UpdateGUIZoom();
+		driver->ClearSystemSprites();
+		ClearFontCache();
+		GfxClearSpriteCache();
+
+		// Apply scaling factor to all windows so when GameSizeChanged is called
+		// from AllocateBackingStore the windows still fit within the main window.
+		float scaling = float(suggestedZoomLevel) / float(newSuggestedZoomLevel);
```

Comment:
This is incorrect as the zoom levels have log scale. I could use `UnScaleByZoom` here to get the scaling factor.

Reply:
Use `-` to find the difference between scaling levels. You probably don't need to use floats for this at all, nothing else does.

Reply:
Indeed, I didn't factor in the possibility of a bit shift. That could work: shifting by the difference.

Reply:
That's exactly what the Zoom and UnZoom functions (`zoom_func.h`) already do.

Reply:
@nielsmh Using `-` would produce [-2, 2] for the current implementation. Is there a method available to apply this zoom difference, or do I need a combination of `ScaleByZoom` and `UnScaleByZoom` to achieve what I want?

Reply:
Yeah you should take the difference between old and new level, if it's negative then use UnScaleByZoom by the absolute value, if it's positive then use ScaleByZoom. You should really read `zoom_func.h`, it's very short and all the functions are documented.

Reply:
Thanks! I’ve updated the code. To reduce duplication I’ve introduced a helper with the condition shown above.